File: runtimecs/java/util/function/Consumer.cs
using java.lang;

namespace java.util.function 
{ 
    public interface Consumer
    {
        void accept(object s);
        Consumer andThen(Consumer other);
    }

    public static class Consumer_0009
    {
        public static Consumer andThen(Consumer @this, Consumer other)
        {   
            if (@this==null || other==null) { throw new NullPointerException(); }
            return new ConsumerAndThen(@this,other);
        }
    }

    public class ConsumerAndThen : Consumer
    {
        private readonly Consumer a;
        private readonly Consumer b;
        
        public ConsumerAndThen(Consumer a, Consumer b) 
        {   
            this.a = a;
            this.b = b;
        }
        public void accept(object o)
        {   
            a.accept(o);
            b.accept(o);
        }
        public virtual Consumer andThen(Consumer other)
        {   
            return Consumer_0009.andThen(this,other);
        }
    }
}        

File: runtimejs/java/util/function/Consumer.js
//reference// java/lang/NullPointerException
//load// java/lang/Object
var java_util_function_Consumer = 
{   _superinterfaces: [], 
    _defaults:
    {   andThen_1: function(after)
        {   if (after===null) 
            {   throw (new java_lang_NullPointerException())._0()._e;
            }
            return new java_util_function_ConsumerAndThen(this,after);
        }
    },
}; 

// methods:
// void accept(T t)
// default Consumer<T> andThen(Consumer<T> after)

var java_util_function_ConsumerAndThen = function(a,b)   
// internal use only - integrate allocator with constructor 
{   this.a = a;
    this.b = b;    
};
_class(java_util_function_ConsumerAndThen, java_lang_Object, [java_util_function_Consumer], 
"java.util.function.ConsumerAndThen"  //replace-me-with-empty-string-for-production//
,{  accept_1: function(o) 
    {   this.a.accept_1(o);
        this.b.accept_1(o);
    },        
});
